### ISSUE TITLE
Bluetooth RTL8723BE not working on Manjaro linux kernel 4.4.8

### ISSUE THREAD

**Issue by Tomahna** [lwfinger/rtlwifi_new#116]
Hi, i can't get bluetooth to work on manjaro with kernel 4.4.8 : 
https://forum.manjaro.org/index.php?topic=33656.0

**Comment by lwfinger** [comment]
I do not have an installation running kernel 4.4, but there should be nothing in the kernel between 4.1 and 4.4 that would break this driver.

If you absolutely need the ant_sel=1 for your device to work, then BT may never work as it may also be trying to use the unattached antenna. I do not understand the BT driver well enough to put the antenna selection fix into it. It seems that functionality is in the firmware which does not come with source. In addition, I have some other matters that are taking my time.

**Comment by Tomahna** [comment]
Well in my case it doesn't work in linux 4.1 nor in linux 4.4, the only thing that change is the nature of the problem. And ant_sel=1 does not make any difference. The only time i've got it to work was with SteamOS where it surprisingly just worked out of the box.

I perfectly understand that you have other things to attend to.

**Comment by lwfinger** [comment]
With my distro (openSUSE), this repo works. As long as the firmware is available, then the problem may be with the Manjaro user-space stuff.

**Comment by Snudl** [comment]
I am on Manjaro aswell and have to this issue (no change from kernel 4.3 to 4.6), though any Arch-based Distro shows the same behaviour, as does Linux Mint. Only on Fedora bluetooth works out of the box, openSUSE I haven't tested yet. Any idea what those are doing differently?

rfkill list - nothing blocked

dmesg | grep -i blue
[    7.003184] Bluetooth: Core ver 2.21
[    7.003197] Bluetooth: HCI device and connection manager initialized
[    7.003199] Bluetooth: HCI socket layer initialized
[    7.003201] Bluetooth: L2CAP socket layer initialized
[    7.003205] Bluetooth: SCO socket layer initialized
[    7.010135] Bluetooth: hci0: rtl: examining hci_ver=06 hci_rev=000b lmp_ver=06 lmp_subver=8723
[    7.010136] Bluetooth: hci0: rtl: loading rtl_bt/rtl8723b_fw.bin
[    7.011126] Bluetooth: hci0: rom_version status=0 version=1
[    7.086063] Bluetooth: BNEP (Ethernet Emulation) ver 1.3
[    7.086064] Bluetooth: BNEP filters: protocol multicast
[    7.086066] Bluetooth: BNEP socket layer initialized
[   13.672835] Bluetooth: RFCOMM TTY layer initialized
[   13.672841] Bluetooth: RFCOMM socket layer initialized
[   13.672845] Bluetooth: RFCOMM ver 1.11
[34569.930126] Bluetooth: hci0 urb ffff88030e7d0900 failed to resubmit (113)
[34579.910283] Bluetooth: hci0 urb ffff88030e7d0d80 failed to resubmit (113)
[34589.897161] Bluetooth: hci0 urb ffff88033cc46d80 failed to resubmit (113)

systemctl status -l bluetooth
● bluetooth.service - Bluetooth service
   Loaded: loaded (/usr/lib/systemd/system/bluetooth.service; enabled; vendor preset: disabled)
   Active: active (running) since Mon 2016-05-30 03:53:26 CEST; 9h ago
     Docs: man:bluetoothd(8)
 Main PID: 581 (bluetoothd)
   Status: "Running"
    Tasks: 1 (limit: 512)
   CGroup: /system.slice/bluetooth.service
           └─581 /usr/lib/bluetooth/bluetoothd

May 30 03:53:26 snu-bot bluetoothd[581]: Bluetooth management interface 1.12 initialized
May 30 03:53:26 snu-bot bluetoothd[581]: Failed to obtain handles for "Service Changed" characteristic
May 30 03:53:26 snu-bot bluetoothd[581]: Sap driver initialization failed.
May 30 03:53:26 snu-bot bluetoothd[581]: sap-server: Operation not permitted (1)
May 30 03:53:35 snu-bot bluetoothd[581]: Endpoint registered: sender=:1.28 path=/MediaEndpoint/A2DPSource
May 30 03:53:35 snu-bot bluetoothd[581]: Endpoint registered: sender=:1.28 path=/MediaEndpoint/A2DPSink
May 30 13:29:36 snu-bot bluetoothd[581]: Failed to set mode: Failed (0x03)
May 30 13:29:46 snu-bot bluetoothd[581]: Failed to set mode: Failed (0x03)

Some systemd weirdness maybe?

**Comment by rickysarraf** [comment]
This issue is also seen on Debian. This should be easily seen with any distro running the stock kernel.

**Comment by lwfinger** [comment]
A completely new version of the drivers has just been pushed. It fixes a lot of things. As a result, I am closing all open issues. If you get a new one, please create a new issue.

Note: The new driver fixes the problems with power save. It is recommended that you clear all ips, swlps, and fwlps options.

**Comment by rickysarraf** [comment]
When you say pushed, should I assume it is part of the Linux kernel 4.9 ?

**Comment by lwfinger** [comment]
Absolutely not. It takes a while for new material to reach mainline. I estimate that it will be 4.11 before all the new code will be there. The code was pushed to THIS repo. None of it is in 4.9.

**Comment by PhilipKigo** [comment]
It works with opensuse after you rfkill unblock all. There are no errors, but i can confirm wasn't working on manjaro.

**Comment by lwfinger** [comment]
I have no control over Manjaro user space. I use openSUSE, and I am very happy with their code.

**Comment by PhilipKigo** [comment]
Thankfully, i use opensuse too. Good job lwfinger.

**Comment by sunitknandi** [comment]
@lwfinger , @Tomahna , @Snudl , @rickysarraf  I have found the exact issue. It is not really a Manjaro problem.
I am running Manjaro, and found the culprit to be "tlp".

A simple
`pacman -Rc tlp tlp-rdw`
and rebooting does the job. Basically you just remove the tlp packages. And voila, BT is working for me.

I know this is not a proper solution for a laptop, but it works for me.

**Comment by Snudl** [comment]
I can confirm that removing tlp on Manjaro fixes the issue. Thanks @sunitknandi

**Comment by PhilipKigo** [comment]
I can confirm that issue and solution in Opensuse tumbleweed

On 1 May 2017 20:47, "Snudl" <notifications@github.com> wrote:

I can confirm that removing tlp on Manjaro fixes the issue. Thanks
@sunitknandi <https://github.com/sunitknandi>

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/lwfinger/rtlwifi_new/issues/116#issuecomment-298385233>,
or mute the thread

**Comment by Tomahna** [comment]
I can confirm it too ! Many Thanks @sunitknandi

**Comment by lobosan** [comment]
Not working for me. When I restart my laptop or if I suspend it, then when comebacks I can't use Bluetooth. Any ideas or guidance would be much appreciated

**Comment by jetojedno** [comment]
Your easiest start is to edit the suspend / resume local scripts to unload the module(s) before suspend and reload them on resume.

It that doesn't work then the hardware will need a reset on resume.

No direct knowledge of this, but experience of similar problems with other HW. :-)

Yours
David
